Remove duplicated files

dongjoon-hyun opened the pull request:
Currently, there are two duplicated files which causes misleading errors on MacOS.

<img width="634" height="76" alt="Screenshot 2025-09-08 at 09 30 53" src="https://github.com/user-attachments/assets/19ea7591-885b-4743-b0ef-0ff7c910851b" />

<img width="626" height="68" alt="Screenshot 2025-09-08 at 09 31 21" src="https://github.com/user-attachments/assets/1cabca0c-c7a6-4727-9568-943dde4099ab" />

dongjoon-hyun committed:
Remove duplicated files

dongjoon-hyun commented:
cc @HyukjinKwon and @viirya

viirya reviewed (commented):
Hmm,

https://spark.apache.org/docs/3.5.6/api/R/reference/isNaN.html is listed as column function (https://spark.apache.org/docs/3.5.6/api/R/reference/columnfunctions.html). But 

https://spark.apache.org/docs/3.5.6/api/R/reference/isnan.html is listed as column non-aggregate function (https://spark.apache.org/docs/3.5.6/api/R/reference/column_nonaggregate_functions.html).

If removing https://spark.apache.org/docs/3.5.6/api/R/reference/isnan.html, does the auto-forwarding work still?

dongjoon-hyun commented:
HTTP protocol is case-insensitive. It will work, @viirya .

In addition, all the other versions (including 3.5.5 and 4.0.1) have no duplications.

dongjoon-hyun commented:
Ah. I got what you meant. You mean it's not a duplication. Did I understand correctly?

https://github.com/apache/spark-website/blob/c3f07a00fe5c6262714505bbf3a2813a7c1081e3/site/docs/3.5.6/api/R/reference/isNaN.html#L1-L7

https://github.com/apache/spark-website/blob/c3f07a00fe5c6262714505bbf3a2813a7c1081e3/site/docs/3.5.6/api/R/reference/isnan.html#L1-L7

dongjoon-hyun commented:
I guess `isnan.html` seems to be added mistakenly because it's a duplication of `isnan%2CColumn-method.html` or `is.nan.html`. Here, @viirya .

https://github.com/apache/spark-website/blob/c3f07a00fe5c6262714505bbf3a2813a7c1081e3/site/docs/3.5.6/api/R/reference/isnan.html#L1-L7

https://github.com/apache/spark-website/blob/c3f07a00fe5c6262714505bbf3a2813a7c1081e3/site/docs/3.5.6/api/R/reference/isnan%2CColumn-method.html#L1-L7

https://github.com/apache/spark-website/blob/c3f07a00fe5c6262714505bbf3a2813a7c1081e3/site/docs/3.5.6/api/R/reference/is.nan.html#L1-L7

dongjoon-hyun commented:
Probably, it might be added mistakenly via some manual Bash script operations.

dongjoon-hyun commented:
Anyway, let's wait for @HyukjinKwon for the confirmation for the correct fix because this happens at 3.5.6 suddenly. Thank you, @viirya .

viirya commented:
> HTTP protocol is case-insensitive. It will work, @viirya .

Yea, I think it is, for example GroupedData/groupedData html.

But seems for is nan, they are linked to different pages, so if you open the two isNaN and isnan html in the browser, they will forward-link to different location. So I wonder if we remove isnan.html, we will break the current link. Although isNaN.html will work and forward link, but it will link to different location.

I'm not sure if this is correct so maybe wait for @HyukjinKwon.

dongjoon-hyun commented:
For `isnan.html` and `isNaN.html`, I confirmed that. 
> But seems for is nan, they are linked to different pages, so if you open the two isNaN and isnan html in the browser, they will forward-link to different location. So I wonder if we remove isnan.html, we will break the current link. Although isNaN.html will work and forward link, but it will link to different location.

I was confused because the HTTP PATH component part is case-sensitive as you pointed out.

```
$ curl https://spark.apache.org/docs/3.5.6/api/R/reference/isnan.html
<html>
  <head>
    <meta http-equiv="refresh" content="0;URL=https://spark.apache.org/docs/3.5.6/api/R/reference/column_nonaggregate_functions.html" />
    <meta name="robots" content="noindex">
    <link rel="canonical" href="https://spark.apache.org/docs/3.5.6/api/R/reference/column_nonaggregate_functions.html">
  </head>
</html>

$ curl https://spark.apache.org/docs/3.5.6/api/R/reference/isNaN.html
<html>
  <head>
    <meta http-equiv="refresh" content="0;URL=https://spark.apache.org/docs/3.5.6/api/R/reference/columnfunctions.html" />
    <meta name="robots" content="noindex">
    <link rel="canonical" href="https://spark.apache.org/docs/3.5.6/api/R/reference/columnfunctions.html">
  </head>
</html>
```

Let me close this PR.

dongjoon-hyun commented:
BTW, how to you handle this on your MacOS? For me, the current `asf-site` branch is very annoying on MacOS because MacOS is case-insensitive.

```
$ git status
On branch asf-site
Your branch is up to date with 'origin/asf-site'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   site/docs/3.5.6/api/R/reference/GroupedData.html
	modified:   site/docs/3.5.6/api/R/reference/isNaN.html
```

viirya commented:
> BTW, how to you handle this on your MacOS? For me, the current `asf-site` branch is very annoying on MacOS because MacOS is case-insensitive.

Hmm, I forgot that if I encountered this before as I didn't checkout `spark-website` for a while. 🤔 
But it looks like an annoying issue. Maybe we should rename one of them to make them separated under case-insensitive environment?
